### ISSUE TITLE
How to display html on external pages of the site?

### ISSUE THREAD

**Issue by karpinskyi** [hlashbrooke/WordPress-Plugin-Template#84]
Could you tell me, in what place occasionally this code, to he earned on all external pages site?
echo '<script>alert("'.get_option('wpt_text_block').'");</script>';

**Comment by karpinskyi** [comment]
I know what a shortcode is, I don't need it. I want to output html via your template.

**Comment by jonathanbossenger** [comment]
@karpinskyi as i explained in your [previous ticket](https://github.com/hlashbrooke/WordPress-Plugin-Template/issues/83), this is not a template in the same way that you would have template files in a WordPress theme, this is just a plugin template, so if you wanted to build a plugin, you could use this as a foundation, and build your required functionality on top of that. But learning how to build that functionality is up to you.

There are many ways to output html into a page, you could use a widget, a shortcode or simply build it into a child theme template. But this is not the place to ask how to learn to do that.

**Comment by karpinskyi** [comment]
You don't understand my question at all. I'm not asking you to teach me how to create a shortcode. Show in what place the files need to create your html (or its functions), which will appear on the pages of the site?

You have a good pattern. I was looking for different ones, but I liked yours the most. But I can't figure out where to edit it for themselves.

**Comment by jonathanbossenger** [comment]
@karpinskyi I do understand your question, that is the point. There is no place in the plugin template where if you add code, it will output to the front page of the site, you would need to create it yourself. 

If you will bear with me for a quick analogy, imagine if building a plugin is like building a house. 

When you build a house, first you build a foundation, then  you build the walls, then the roof, then you install the windows, doors, and all the fittings. 

This plugin is like the foundation of your house, but you still need to learn how to build the walls and the roof and install the windows and doors and fittings. 

You're asking me to show you how to install a window, but you haven't built any walls yet.

**Comment by karpinskyi** [comment]
@jonathanbossenger You made a great admin structure. All I need is to change the input fields for myself. It's very simple, I figured it out in 5 minutes.
Are you saying you don't have the outside of the site? class-wordpress-plugin-template.php It seemed to me that this file is responsible for the external part of the site. 

I understand what it means to build a plugin. But I can't figure out where to push off. I could create my own file and php class, but how do I connect it correctly to your template?

I want to build some plugins based on your template.
I found your pattern in this ranking. Your version seemed to me the most successful.https://includewp.com/
Tell me, Where do I start to build the outer part?
Thank You.
